include/zephyr/pm/state.h

```diff
@@ -154,6 +154,14 @@ struct pm_state_info {
 	 * @note 0 means that this property is not available for this state.
 	 */
 	uint32_t exit_latency_us;
+
+#ifdef CONFIG_PM_STATE_CUSTOM_DATA
```

Comment:
Thinking more about it I believe it is a bad idea, at least for this case. The data is quite dependent of the target and the way it is being used in SoF ended up calling target specific API (` intel_adsp_acquire_clock_switch_lock`).
It does not make sense having this workaround if you need to call that API anyways.